test/compilable/test16492.d

```diff
@@ -13,3 +13,26 @@ void test() @nogc pure
         b ~= 4;
     }
 }
+
+void debugSafe() @safe
+{
+    debug unsafe();
+    debug unsafe2();
+}
+
+void unsafe() @system {}
+void unsafe2()() {
+    int[] arr;
+    auto b = arr.ptr;
+}
+
+void debugSafe2() @safe
+{
+    char[] arr1, arr2;
+    debug unsafe3(arr1, arr2);
```

Comment:
I meant something like:
```D
void debugSafe2() @safe
{
    char*ptr;
    char[] arr;
    debug ptr = arr.ptr;
}
```

Reply:
Ah sorry. Yeah that was indeed still forbidden. Fixed.